modules/broker-amqp/src/main/java/io/ballerina/messaging/broker/amqp/codec/handlers/AmqpConnectionHandler.java

```diff
@@ -286,14 +286,25 @@ public int getId() {
 
     /**
      * Sends a connection close frame to the client.
-     * @param reason reason to disconnect connection
+     * @param reason reason to close connection
      */
-    public void forceDisconnect(String reason) {
+    public void closeConnection(String reason) {
         ctx.writeAndFlush(new ConnectionClose(AmqConstant.CONNECTION_FORCED,
                                               ShortString.parseString("Broker forced close connection. " + reason),
                                               0, 0));
     }
 
+    /**
+     * Closes the underlying connection with the client.
+     *
+     * @param reason reason to force close connection
+     * @return int representing the number of channels registered on the connection
+     */
+    public int forceCloseConnection(String reason) {
+        ctx.close();
+        return channels.size();
```

Comment:
Shall we add an info log for the reason? This wouldn't be a thing that we do regularly hence info level is fine I think. And it would be better if we can log the number of channels it had and the connection id as well 

Something similar to the following will help. With this, we get the error trace if an error happens as well.

```java
        ChannelFuture close = ctx.close();
        close.addListener(future -> {
            if (future.isSuccess()) {
                LOGGER.info("Connection forcefully disconnected. Reason: {}", reason);
            } else {
                LOGGER.error("Error occurred while closing connection with id: {}", getId(), future.cause());
            }
        });
```